test/compliance/generate_test_stft_data.py

```diff
@@ -1,80 +1,69 @@
+def run(exe_path, scp_path, out_dir, wave_len, num_outputs, verbose):
+    for i in range(num_outputs):
         inputs = {
             'blackman_coeff': '%.4f' % (random.random() * 5),
             'dither': '0',
             'energy_floor': '%.4f' % (random.random() * 5),
-            'frame_length': '%.4f' % (float(random.randint(2, WAV_LEN - 1)) / 16000 * 1000),
-            'frame_shift': '%.4f' % (float(random.randint(1, WAV_LEN - 1)) / 16000 * 1000),
+            'frame_length': '%.4f' % (float(random.randint(2, wave_len - 1)) / 16000 * 1000),
+            'frame_shift': '%.4f' % (float(random.randint(1, wave_len - 1)) / 16000 * 1000),
             'preemphasis_coefficient': '%.2f' % random.random(),
-            'raw_energy': generate_rand_boolean(),
-            'remove_dc_offset': generate_rand_boolean(),
-            'round_to_power_of_two': generate_rand_boolean(),
-            'snip_edges': generate_rand_boolean(),
-            'subtract_mean': generate_rand_boolean(),
-            'window_type': generate_rand_window_type()
+            'raw_energy': utils.generate_rand_boolean(),
+            'remove_dc_offset': utils.generate_rand_boolean(),
+            'round_to_power_of_two': utils.generate_rand_boolean(),
+            'snip_edges': utils.generate_rand_boolean(),
+            'subtract_mean': utils.generate_rand_boolean(),
+            'window_type': utils.generate_rand_window_type()
         }
 
         fn = 'spec-' + ('-'.join(list(inputs.values())))
 
-        arg = [
-            EXE_PATH,
-            '--blackman-coeff=' + inputs['blackman_coeff'],
-            '--dither=' + inputs['dither'],
-            '--energy-floor=' + inputs['energy_floor'],
-            '--frame-length=' + inputs['frame_length'],
-            '--frame-shift=' + inputs['frame_shift'],
-            '--preemphasis-coefficient=' + inputs['preemphasis_coefficient'],
-            '--raw-energy=' + inputs['raw_energy'],
-            '--remove-dc-offset=' + inputs['remove_dc_offset'],
-            '--round-to-power-of-two=' + inputs['round_to_power_of_two'],
-            '--sample-frequency=16000',
-            '--snip-edges=' + inputs['snip_edges'],
-            '--subtract-mean=' + inputs['subtract_mean'],
-            '--window-type=' + inputs['window_type'],
-            SCP_PATH,
-            OUTPUT_DIR + fn + '.ark'
-        ]
+        out_fn = out_dir + fn + '.ark'
+
+        arg = [exe_path]
+        arg += ['--' + k.replace('_', '-') + '=' + inputs[k] for k in inputs]
+        arg += [scp_path, out_fn]
 
         print(fn)
         print(inputs)
         print(' '.join(arg))
 
         try:
-            subprocess.call(arg)
+            if verbose:
+                subprocess.call(arg)
+            else:
+                subprocess.call(arg, stderr=open(os.devnull, 'wb'), stdout=open(os.devnull, 'wb'))
+            print('success')
         except Exception:
-            pass
+            if os.path.exists(out_fn):
```

Comment:
I wouldn't remove things by default. If someone runs the command by accident this can be very frustrating.